pandas/core/resample.py

```diff
@@ -1797,3 +1795,28 @@ def asfreq(obj, freq, method=None, how=None, normalize=False, fill_value=None):
             new_obj.index = new_obj.index.normalize()
 
     return new_obj
+
+
+def _asfreq_compat(index, freq):
```

Comment:
why can we not just .asfreq(...)? this seems very odd to a) put here (its an Index method), and b) why doesn't this wrok?

Reply:
there is no DatetimeIndex.asfreq or TimedeltaIndex.asfreq (and shouldnt be)